src/kinetics/LinearBurkeRate.cpp

```diff
@@ -0,0 +1,354 @@
+}
+
+double LinearBurkeRate::evalTroeRate(const LmrData& shared_data, DataTypes& dataObj, RateTypes& rateObj)
+{
+    FalloffData& data = boost::get<FalloffData>(dataObj);
+    TroeRate& rate = boost::get<TroeRate>(rateObj);
+    data.conc_3b = {exp(logPeff_) / GasConstant / shared_data.temperature};
+    data.logT = shared_data.logT;
+    data.recipT = shared_data.recipT;
+    data.temperature = shared_data.temperature;
+    rate.setRateIndex(0);
+    return rate.evalFromStruct(data);
+}
+
+double LinearBurkeRate::evalChebyshevRate(const LmrData& shared_data, DataTypes& dataObj, RateTypes& rateObj)
+{
+    ChebyshevData& data = boost::get<ChebyshevData>(dataObj);
+    ChebyshevRate& rate = boost::get<ChebyshevRate>(rateObj);
+    data.log10P = log10(exp(logPeff_));
+    data.logT = shared_data.logT;
+    data.recipT = shared_data.recipT;
+    rate.updateFromStruct(data);
+    return rate.evalFromStruct(data);
+}
+
+double LinearBurkeRate::evalFromStruct(const LmrData& shared_data)
+{
+    double sigmaX_M = 0.0;
+    // Test each species listed at the top of the YAML file
+    for (size_t i = 0; i<nSpecies; i++){
+        // Total sum will be essentially 1, but perhaps not exactly due to Cantera's rounding conventions
+        sigmaX_M += shared_data.moleFractions[i];
+    }
+    eps_mix = 0.0;
+    size_t counter = 0;
+    // Break loop when all colliders have been located to avoid unnecessary iterations
+    while(counter<colliderIndices.size()){
+        // Test each species listed at the top of the YAML file
+        for (size_t i = 0; i<nSpecies; i++){
+            for (size_t j = 0; j<colliderIndices.size(); j++){
+                // Check if the species is specified as an LMR-R collider
+                if (i == colliderIndices[j]) {
+                    eps_mix += shared_data.moleFractions[i] * epsObjs1[j].evalRate(shared_data.logT, shared_data.recipT);
+                    sigmaX_M -= shared_data.moleFractions[i];
+                    counter += 1;
+                     // Break after collider has been located to avoid unnecessary iterations
+                    break;
+                }
+            }
+        }
+    }
+    double eps_M = epsObj_M.evalRate(shared_data.logT, shared_data.recipT);
+    // Add all M colliders to eps_mix in a single step
+    eps_mix += sigmaX_M * eps_M;
+    if (eps_mix == 0) {
+        throw InputFileError("LinearBurkeRate::evalFromStruct", m_input, "eps_mix == 0 for some reason");
+    }
+    double k_LMR_ = 0.0;
+    counter = 0;
+    while(counter<colliderIndices.size()){
+        for (size_t i = 0; i < nSpecies; i++){
+            for (size_t j = 0; j < colliderIndices.size(); j++){
+                if (i == colliderIndices[j]) {
+                    double eps1 = epsObjs1[j].evalRate(shared_data.logT, shared_data.recipT);
+                    double eps2 = epsObjs2[j].evalRate(shared_data.logT, shared_data.recipT);
+                    // eps2 equals either eps_M or eps_i, depending on the scenario
```

Comment:
This loop structure is far more complicated than necessary. You can reduce this to
```suggestion
    for (size_t j = 0; j < colliderIndices.size(); j++) {
        size_t i = colliderIndices[j];
        double eps1 = epsObjs1[j].evalRate(shared_data.logT, shared_data.recipT);
        double eps2 = epsObjs2[j].evalRate(shared_data.logT, shared_data.recipT);
        // eps2 equals either eps_M or eps_i, depending on the scenario
```
and eliminate the `counter` variable and the `break` statements after evaluating the corresponding rate.